cmd/ndm_daemonset/probe/seachestprobe.go

```diff
@@ -124,65 +124,65 @@ func (scp *seachestProbe) FillBlockDeviceDetails(blockDevice *blockdevice.BlockD
+		klog.V(4).Infof("Disk: %s TotalBytesRead:%d filled by seachest.", blockDevice.DevPath, blockDevice.SMARTInfo.TotalBytesRead)
 	}
 
-	if d.TotalBytesWritten == 0 {
-		d.TotalBytesWritten = seachestProbe.SeachestIdentifier.GetTotalBytesWritten(driveInfo)
-		klog.V(4).Infof("Disk: %s TotalBytesWritten:%d filled by seachest.", d.Path, d.TotalBytesWritten)
+	if blockDevice.SMARTInfo.TotalBytesWritten == 0 {
+		blockDevice.SMARTInfo.TotalBytesWritten = seachestProbe.SeachestIdentifier.GetTotalBytesWritten(driveInfo)
+		klog.V(4).Infof("Disk: %s TotalBytesWritten:%d filled by seachest.", blockDevice.DevPath, blockDevice.SMARTInfo.TotalBytesWritten)
 	}
 
-	if d.DeviceUtilizationRate == 0 {
-		d.DeviceUtilizationRate = seachestProbe.SeachestIdentifier.GetDeviceUtilizationRate(driveInfo)
-		klog.V(4).Infof("Disk: %s DeviceUtilizationRate:%f filled by seachest.", d.Path, d.DeviceUtilizationRate)
+	if blockDevice.SMARTInfo.UtilizationRate == 0 {
+		blockDevice.SMARTInfo.UtilizationRate = seachestProbe.SeachestIdentifier.GetDeviceUtilizationRate(driveInfo)
+		klog.V(4).Infof("Disk: %s UtilizationRate:%f filled by seachest.", blockDevice.DevPath, blockDevice.SMARTInfo.UtilizationRate)
 	}
 
-	if d.PercentEnduranceUsed == 0 {
-		d.PercentEnduranceUsed = seachestProbe.SeachestIdentifier.GetPercentEnduranceUsed(driveInfo)
-		klog.V(4).Infof("Disk: %s PercentEnduranceUsed:%f filled by seachest.", d.Path, d.PercentEnduranceUsed)
-	}*/
+	if blockDevice.SMARTInfo.PercentEnduranceUsed == 0 {
+		blockDevice.SMARTInfo.PercentEnduranceUsed = seachestProbe.SeachestIdentifier.GetPercentEnduranceUsed(driveInfo)
+		klog.V(4).Infof("Disk: %s PercentEnduranceUsed:%f filled by seachest.", blockDevice.DevPath, blockDevice.SMARTInfo.PercentEnduranceUsed)
+	}
 
-	blockDevice.TemperatureInfo.TemperatureDataValid = seachestProbe.
+	blockDevice.SMARTInfo.TemperatureInfo.CurrentTemperatureDataValid = seachestProbe.
 		SeachestIdentifier.GetTemperatureDataValidStatus(driveInfo)
 	klog.V(4).Infof("Disk: %s TemperatureDataValid:%t filled by seachest.",
-		blockDevice.DevPath, blockDevice.TemperatureInfo.TemperatureDataValid)
+		blockDevice.DevPath, blockDevice.SMARTInfo.TemperatureInfo.CurrentTemperatureDataValid)
 
-	if blockDevice.TemperatureInfo.TemperatureDataValid == true {
-		blockDevice.TemperatureInfo.CurrentTemperature = seachestProbe.
+	if blockDevice.SMARTInfo.TemperatureInfo.CurrentTemperatureDataValid == true {
+		blockDevice.SMARTInfo.TemperatureInfo.CurrentTemperature = seachestProbe.
 			SeachestIdentifier.GetCurrentTemperature(driveInfo)
 
 		klog.V(4).Infof("Disk: %s CurrentTemperature:%d filled by seachest.",
-			blockDevice.DevPath, blockDevice.TemperatureInfo.CurrentTemperature)
+			blockDevice.DevPath, blockDevice.SMARTInfo.TemperatureInfo.CurrentTemperature)
 
-		/*d.TemperatureInfo.HighestValid = seachestProbe.
+		blockDevice.SMARTInfo.TemperatureInfo.HighestTemperatureDataValid = seachestProbe.
```

Comment:
Can the check be made for highest and lowest temperature data also. i.e fill the highest/lowest temp only if the corresponding field is valid. @harshthakur9030